Source/objects.cpp

```diff
@@ -1158,94 +1158,62 @@ void InitObjects()
 
 void SetMapObjects(unsigned char *pMap, int startx, int starty)
 {
-	unsigned char *v3;  // esi
-	int v6;             // edi
-	int v7;             // eax
-	int v8;             // esi
-	int v9;             // ecx
-	int v10;            // esi
-	int v11;            // ecx
-	int v12;            // edi
-	_BYTE *v13;         // eax
-	int v14;            // ebx
-	signed int v15;     // ebx
-	char *v16;          // ST08_4
-	unsigned char *v17; // eax
-	int v18;            // ecx
-	int i;              // ebx
-	int fileload[56];   // [esp+Ch] [ebp-10Ch]
-	char filestr[32];   // [esp+ECh] [ebp-2Ch]
-	_BYTE *v22;         // [esp+10Ch] [ebp-Ch]
-	int v23;            // [esp+110h] [ebp-8h]
-	_BYTE *v24;         // [esp+114h] [ebp-4h]
-	int y;              // [esp+120h] [ebp+8h]
-
-	v23 = startx;
-	v3 = pMap;
+	int rw, rh;
+	int i, j;
+	unsigned char *lm, *h;
+	long mapoff;
+	int fileload[56];
+	char filestr[32];
+
 	ClrAllObjects();
-	memset(fileload, 0, sizeof(fileload));
+	for (i = 0; i < 56; i++)
+		fileload[i] = 0;
 	InitObjFlag = TRUE;
-	if (AllObjects[0].oload != -1) {
-		i = 0;
-		do {
-			if (AllObjects[i].oload == 1 && leveltype == AllObjects[i].olvltype)
-				fileload[AllObjects[i].ofindex] = 1;
-			i++;
-		} while (AllObjects[i].oload != -1);
-	}
-	v6 = (unsigned char)*v3;
-	v7 = (int)(v3 + 2);
-	v8 = (unsigned char)v3[2];
-	v9 = v8;
-	v10 = 2 * v8;
-	v11 = v6 * v9;
-	v12 = 2 * v6;
-	v13 = (_BYTE *)(2 * v11 + 2 + 4 * v12 * v10 + v7);
-	v22 = v13;
-	if (v10 > 0) {
-		v24 = (_BYTE *)v10;
-		do {
-			if (v12 > 0) {
-				v14 = v12;
-				do {
-					if (*v13)
-						fileload[(char)AllObjects[ObjTypeConv[(unsigned char)*v13]].ofindex] = 1;
-					v13 += 2;
-					--v14;
-				} while (v14);
+
+	for (i = 0; AllObjects[i].oload != -1; i++) {
+		if (AllObjects[i].oload == 1 && leveltype == AllObjects[i].olvltype)
+			fileload[AllObjects[i].ofindex] = 1;
+	}
+
+	lm = pMap + 2;
+	rh = pMap[0];
+	rw = *lm;

```

Comment:
`rw` and `rh` should be swapped. The first two bytes are width and next two height. Also, you can clean this up as so:
```c
lm = pMap;
rw = *pMap;
lm += 2;
rh = *pMap;
```